### PR TITLE
add existing tag-album and tag-track check in load script, change tests accordingly

### LINKED ISSUE #127: `load.py` - handle duplicate insertions
Edit the `load.py` script to handle duplicate insertions into the `album_tag_assignment' table as well as the 'track_tag_assignment` table.
Do this by checking if an assignment already exists before inserting.

As engineers we need to ensure that the duplicate insertions are not taking place as this slows down the queries unnecessarily

### PR DESCRIPTION
**Pull request by e-lemma**
I have added code to ensure any existing album/track to tag assignments are found and acknowledged, instead of being erroneously added again to the database. I have also changed the tests to reflect these new changes.
All tests pass, all pylint scores above 8
Resolves #127

### CODE REVIEW

File: pipeline/load.py [hunk 1]
```diff
@@ -148,16 +151,30 @@ def get_or_insert_tags(
...
-            "INSERT INTO album_tag_assignment(tag_id, album_id) VALUES (%s, %s)",
-            (tag_id, album_id),
+            "SELECT album_tag_assignment_id from album_tag_assignment WHERE album_id = %s and tag_id = %s",
+            (album_id, tag_id),
         )
+        ata_id = cursor.fetchone()
+        if not ata_id:
+            cursor.execute(
+                "INSERT INTO album_tag_assignment(tag_id, album_id) VALUES (%s, %s)",
+                (tag_id, album_id),
```
Comment (alina-101) [inline pipeline/load.py:165]: Perfect, this will mean no duplicates!

File: pipeline/load.py [hunk 2]
```diff
@@ -272,7 +289,7 @@ def insert_single_sale(cursor: DBCursor, single_sale: Dict[str, Any]) -> None:
 
     for tag in single_sale["track_tags"]:
         if tag:
-            get_or_insert_tags(cursor, tag, track_id=track_id)
+            get_or_insert_tags_and_assignments(cursor, tag, track_id=track_id)
```
Comment (alina-101) [inline pipeline/load.py:292]: could this just be `track_id`

File: pipeline/load.py (hunk 1 above)
Comment (e-lemma) [reply pipeline/load.py:165]: I hope so

File: pipeline/load.py (hunk 2 above)
Comment (e-lemma) [reply pipeline/load.py:292]: This function is used for both albums and tracks, and so can take in both album_id and track_id as arguments. Both these arguments are `None` by default, and the specific target of the function is specified by mentioning the specific argument, such as `track_id=track_id`. This method of specifying the argument is used when there are more than one arguments with default values. 
You can read more about this [here](https://stackoverflow.com/questions/25374814/multiple-default-values-for-function-arguments-python)

Comment (alina-101) [reply pipeline/load.py:292]: That makes sense! Good explanation

**Review by alina-101 — APPROVED** [review]
Looks good. Great that we can now avoid duplicates!

**Pull request merged**